File: backend/app/schemas/event.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class EventResponse(BaseModel):
    id: uuid.UUID
    camera_id: uuid.UUID
    camera_name: str | None = None
    event_type: str
    severity: str
    confidence: float
    thumbnail_path: str | None
    clip_path: str | None
    metadata_json: dict | None
    is_acknowledged: bool
    acknowledged_by: uuid.UUID | None
    acknowledged_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class EventListParams(BaseModel):
    camera_id: uuid.UUID | None = None
    event_type: str | None = None
    severity: str | None = None
    start_date: datetime | None = None
    end_date: datetime | None = None
    is_acknowledged: bool | None = None


class AcknowledgeRequest(BaseModel):
    event_ids: list[uuid.UUID]

File: frontend/src/types/event.ts
export interface Event {
  id: string;
  camera_id: string;
  camera_name: string | null;
  event_type: string;
  severity: string;
  confidence: number;
  thumbnail_path: string | null;
  clip_path: string | null;
  metadata_json: Record<string, unknown> | null;
  is_acknowledged: boolean;
  acknowledged_by: string | null;
  acknowledged_at: string | null;
  created_at: string;
}

export interface EventListParams {
  camera_id?: string;
  event_type?: string;
  severity?: string;
  start_date?: string;
  end_date?: string;
  is_acknowledged?: boolean;
  limit?: number;
  offset?: number;
}

export interface AcknowledgeRequest {
  event_ids: string[];
}
